### ISSUE TITLE
Add support for data_actions in azurerm_role_definition

### ISSUE THREAD

**Issue by Supermathie** [terraform-providers/terraform-provider-azurerm#1538]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I want to make a role definition that is only allows to perform data actions on blobs within containers, but `azurerm_role_definition` does not have support for specifying `data_actions`.

For reference here is the existing `'Storage Blob Data Contributor (Preview)'` role provided by Azure:

```
 'permissions': [{'actions': ['Microsoft.Storage/storageAccounts/blobServices/containers/delete',
                              'Microsoft.Storage/storageAccounts/blobServices/containers/read',
                              'Microsoft.Storage/storageAccounts/blobServices/containers/write'],
                  'dataActions': ['Microsoft.Storage/storageAccounts/blobServices/containers/blobs/delete',
                                  'Microsoft.Storage/storageAccounts/blobServices/containers/blobs/read',
                                  'Microsoft.Storage/storageAccounts/blobServices/containers/blobs/write'],
                  'notActions': [],
                  'notDataActions': []}],
```

### New or Affected Resource(s)

* azurerm_role_definition

### Potential Terraform Configuration

```hcl
resource "azurerm_role_definition" "blob_rw" {
  name               = "Blob Read/Write"
  …
  permissions {
    actions     = [
        'Microsoft.Storage/storageAccounts/blobServices/containers/read',
    ]
    data_actions = [
      "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/read",
      "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/write",
    ]
    not_actions = []
    not_data_actions = [
      "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/delete",
    ]
  }
  …
}
```

### References

* https://docs.microsoft.com/en-us/azure/role-based-access-control/role-definitions

**Comment by jondkent** [comment]
Any chance this is going to get looked at soon?

**Comment by tombuildsstuff** [comment]
hey @Supermathie @jondkent 

Just to let you know that support for this has now landed in v1.16 of the AzureRM Provider (here's the full changelog](https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.16.0/CHANGELOG.md)). You can upgrade to to this version by specifying it in your provider block:

```
provider "azurerm" {
  version = "=1.16.0"
}
```

... and then running `terraform init -upgrade` :)

Thanks!
